From 7673cece194e846cb0b68569178252ff16e9655e Mon Sep 17 00:00:00 2001
From: David Milburn <dmilburn@redhat.com>
Date: Mon, 18 Aug 2014 19:58:41 -0400
Subject: [block] nvme: Enable BUILD_BUG_ON checks

Message-id: <1408391935-24886-9-git-send-email-dmilburn@redhat.com>
Patchwork-id: 87889
O-Subject: [RHEL7.1 PATCH BZ 1111259 08/22] NVMe: Enable BUILD_BUG_ON checks
Bugzilla: 1111259
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Matthew Wilcox <matthew.r.wilcox@intel.com>

Since _nvme_check_size() wasn't being called from anywhere, the compiler
was optimising it away ... along with all the link-time build failures
that would result if any of the structures were the wrong size.  Call it
from nvme_exit() for no particular reason.

Signed-off-by: Matthew Wilcox <matthew.r.wilcox@intel.com>
(cherry picked from commit 21bd78bcf4208e84deab0d34f9d4e034d0580d0c)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/block/nvme-core.c b/drivers/block/nvme-core.c
index 8307982..0c34682 100644
--- a/drivers/block/nvme-core.c
+++ b/drivers/block/nvme-core.c
@@ -3000,6 +3000,7 @@ static void __exit nvme_exit(void)
  unregister_blkdev(nvme_major, "nvme");
  destroy_workqueue(nvme_workq);
  BUG_ON(nvme_thread && !IS_ERR(nvme_thread));
+ _nvme_check_size();
 }
 
 MODULE_AUTHOR("Matthew Wilcox <willy@linux.intel.com>");
-- 
1.7.1